### ISSUE TITLE
Support for Allure Reporter

### ISSUE THREAD

**Issue by samuelvinay91** [pactumjs/pactum-examples#12]
Hi @ASaiAnudeep,

We are using allure reporter across our org. We were exploring API test tools and found this awesome tool which seems to be quite feature-rich compared to supertest. 

We need your help in using allure reporting for our POC

**Comment by samuelvinay91** [comment]
Hi @ASaiAnudeep 

I was able to solve it in the following way.Maybe you can correct me or make it better

```js
const allure_reporter = {
  afterSpec() {
    allure.attachment(
      "Request :",
      JSON.stringify(spec.request, null, 2),
      ContentType.JSON
    );
    allure.attachment(
      "Response :",
      JSON.stringify(spec.response, null, 2),
      ContentType.JSON
    );
  },
};

describe("AUDIT LOG", function () {
  let _spec = spec();

  before(async () => {
    decorate<MochaAllure>(allure);
    await request.setBaseUrl(process.env.QA_BASE_URI_V1);
    await reporter.add(allure_reporter);
  });
  ```

**Comment by ASaiAnudeep** [comment]
Hello @SamuelVinay91 ,

The example you have provided looks good to me. Can you raise a PR to add allure example similar to https://github.com/pactumjs/pactum-examples/tree/main/reporter-mochawesome?

**Comment by leelaprasadv** [comment]
@SamuelVinay91 You even try this npm package from allure framework. If you are using mocha/jest/cucumberjs it can be used via cli parameter or api

https://github.com/allure-framework/allure-js

**Comment by samuelvinay91** [comment]
> Hello @SamuelVinay91 ,
> 
> The example you have provided looks good to me. Can you raise a PR to add allure example similar to https://github.com/pactumjs/pactum-examples/tree/main/reporter-mochawesome?

Sure @ASaiAnudeep - I am happy to.

**Comment by codewithmmak** [comment]
@ASaiAnudeep - Implemented the Allure report in PR #13. Please assigned the issue to me to close it.
